### ISSUE TITLE
TS erroring out on seemingly valid <Link /> url

### ISSUE THREAD

**Issue by arackaf** [TanStack/router#2487]
### Which project does this relate to?

Router

### Describe the bug

The `to` value is being reported by TS as an invalid route here

```ts
<Link to="/app/" className="[&.active]:font-bold">
```

even though I do have an index.tsx route inside of `/app` which has a path defined like this 

```ts
export const Route = createFileRoute("/app/")({
```

See below for complete repro steps 

### Your Example Website or App

https://github.com/arackaf/tanstack-router-loader-demo/tree/special/router-bug

### Steps to Reproduce the Bug or Issue

Repo is here (note the branch!) https://github.com/arackaf/tanstack-router-loader-demo/tree/special/router-bug 

This is the line that errors out https://github.com/arackaf/tanstack-router-loader-demo/blob/special/router-bug/src/routes/app/route.tsx#L22 

### Expected behavior

Route above appears valid, and should be accepted 

### Screenshots or Videos

n/a

### Platform

- OS: MacOS
- Browser: n/a
- Version: n/a

**Comment by schiller-manuel** [comment]
this is most likely caused by the `trailingSlash` property 


https://tanstack.com/router/latest/docs/framework/react/api/router/RouterOptionsType#trailingslash-property

since the default is `never` this behavior is expected

**Comment by arackaf** [comment]
Doesn't that property control how the browser treats trailing slashes on routes? Are you saying it also controls how the type system treats then in navigation url's? 

Assuming the trailing slash is set to never, by default, what would be the correct value of that Link's `to` property to work as expected?

**Comment by schiller-manuel** [comment]
> Are you saying it also controls how the type system treats then in navigation url's?

yes

> what would be the correct value of that Link's to property to work as expected?

`<Link to="/app" className="[&.active]:font-bold">`

**Comment by arackaf** [comment]
@schiller-manuel thank you SO MUCH for taking the time, here. I've figured out what the issue is.

I had `/app` initially. But I noticed the Link was marked as active even though I was on child routes. I assumed the solution was to add the trailing slash, in order to link directly to the page itself. But I've just now taken a look at the <Link options and discovered `activeOptions={{ exact: true }}` which solves my issue. 

So sorry about that - honestly it just never occured to me to even look for that. I thought for sure I should be linking to the page with the trailing slash, since that's the route path that's generated by the tanstack cli.
